### ISSUE TITLE
Allow for user-based customization

### ISSUE THREAD

**Issue by thewilli** [gitpod-io/gitpod#4862]
The `.gitpod.yml` file allows to specify **required** extensions. But what about optional ones?

I'm working with a team on a project where some users like me do rely on the [vim keybindings](https://open-vsx.org/extension/vscodevim/vim), while others are happy with the default configuration. The same applies to VSCode *settings* that are sometimes customized in the same way for every project. Just consider you got a new computer for work with a vanilla installation of your IDE of choice. Isn't there some some tuning, aren't some customizations you will definitely do prior to writing the first line of code? 

tl;dr I do love the idea of `gitpod`, but I am really missing some degree of customization to make it feel like *my* IDE.

My proposal is to extend the dashboard to support

* optional extensions which get installed in addition to those listed in `.gitpod.yml`
* additional VSCode settings

**Comment by jankeromnes** [comment]
Hi @thewilli! Many thanks for this great feedback.

You're right that your project's `.gitpod.yml` allows specifying which extensions should get installed automatically in every workspace, for everyone working on your project.

But there are also a few additional ways to install or recommend extensions that might address (parts of) your use cases. I'll list them below, but please feel free to point out what doesn't work for you / what you'd like to achieve that isn't covered by these options.

1. Manually installing a "synced" extension

When you open the Extensions panel in Gitpod's IDE, you can search for an extension, and install it in "synced" mode. This mechanism should save that extension in your user profile, and then install it into every new Gitpod workspace that you create (independently of which repository or what's in the `.gitpod.yml`).

2. Committing a `.vscode/extensions.json` in your repository

Since Gitpod is actually running VS Code in your browser, you can commit a `.vscode/extensions.json` file into your project, and for example recommend various extensions to your team members / contributors: Learn more about [recommending extensions for your project](https://tattoocoder.com/recommending-vscode-extensions-within-your-open-source-projects/)

3. Committing a `.vscode/settings.json`

Similarly, for other VS Code settings, you can either synchronize settings across all your (personal) workspaces, or commit a `.vscode/settings.json` file into your project for the benefit of your team members and contributors. Learn more about [User and Workspace Settings](https://code.visualstudio.com/docs/getstarted/settings)

Would this help address your expectations? Or would you want some other way to customize Gitpod for yourself & others, e.g. from the dashboard (and how would you expect that UX to work / look like)?

**Comment by csweichel** [comment]
Thank you jan for your detailed response. I'll close this issue for now. Please re-open if the answer did not address your expectations.
